rfd/0005/README.mdx

```diff
@@ -0,0 +1,453 @@
+- Semantic versioning for all releases
+- Clear upgrade path documentation
+- Version cleanup recommendations
+- Avoid using "latest" in production environments
+- Regular cleanup of unused versions
+- Clear distinction between Aurora and Moose installation patterns
+- Use language-specific package managers for library dependencies
+
+### Tutorials and Examples
+
+Example workflow:
+```bash
+# Install Aurora (unchanged)
+curl -sSL https://www.fiveonefour.com/install.sh | sh -s aurora
+
+# Install Moose wrapper
+curl -sSL https://www.fiveonefour.com/install.sh | sh -s moose
+
+# Create new project (automatically installs required version)
+moose init my-project
+
+# Libraries are installed via language package managers
+npm install  # For TypeScript projects
+pip install -r requirements.txt  # For Python projects
+
+# Switch to project with different version (automatic)
+cd other-project
+moose build  # Uses version from project config
+
+# Get absolute path for MCP
+moose --binary-path  # Returns absolute path to current version
+```
+
+### User Impact
+
+- Existing projects: No immediate impact
+- New projects: Automatic version management
+- Migration: Optional, can continue using fixed versions
+- Rollout: Gradual, with opt-in for version management
+- Unified installation experience across languages
+- No impact on Aurora users (maintains existing behavior)
+- Libraries continue to be managed by language package managers
+
+## Implementation Plan
+
+### Phase 0: Minimal Version Management (v0)
+Goal: Basic version management with complete channel support and backward compatibility
+
+1. Basic Wrapper Script:
+   - Install wrapper script via existing install.sh
+   - Support `moose` vs `aurora` installation modes
+   - Basic version detection from project files
+   - Use latest version outside projects
+   - Simple version caching in ~/.moose/versions
+
+2. Channel Support:
+   - Full stable/dev channel implementation from day one
+   - Support for git branch versions in dev channel
+   - Automatic version detection from both channels
+   - Channel selection in project configuration
+
+3. Essential Commands:
+   ```bash
+   moose install <version>  # Install specific version
+   moose install latest    # Install latest from stable
+   moose install latest --channel dev  # Install latest from dev
+   moose install @branch-name --channel dev  # Install from git branch
+   ```
+
+4. Backward Compatibility:
+   - Support existing project structures
+   - Handle legacy version specifications
+   - Maintain existing CLI behaviors during transition
+
+5. Minimal Project Integration:
+   - Read versions from package.json/requirements.txt
+   - Basic version switching based on project
+   - Use latest for moose init
+   - Channel specification in project files
+
+### Phase 1: Enhanced Version Management
+Goal: Improve version management reliability and user experience
+
+1. Version Cache Management:
+   - Implement version limit (100)
```

Comment:
I was trying to see how big the binary is. It looks like around 25MB? https://github.com/514-labs/moose/releases/tag/v0.4.89

100 would be around 2.5GB. Could that be too much (assuming binary size grow over time)?

Reply:
Maybe?

I was thinking that it would be configurable so that people want less, they can have less.